### PR TITLE
Remove `ethereumj` dependency from `hedera-services`

### LINKED ISSUE #3077: Eliminate `ethereumj` dependency
### Problem

Although we have `com.esaulpaugh:headlong` on the classpath, and hence no longer need to use `com.hedera.hashgraph:ethereumj-core` for ABI encoding/decoding in _test-clients/_, we have not completed this refactor.

Especially because `ethereumj-core` has a transitive Spring dependency, this adds unnecessary noise to our classpath.

### Solution

Refactor all ABI encoding/decoding under _test-clients/_ to use headlong. Remove `ethereumj-core` dependency.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by IvanKavaldzhiev**
**Description**:
This PR refactors all encoding/decoding logic in test-clients to be performed via headlong library. Thus, we can remove 
ethereumj dependency from the whole project. The changes impact only test-clients module.

This PR deletes `ContractMintHTSSuite.fungibleTokenMintFailure()` since it's usecase was passing invalid argument value that fails the encoding phase when the transaction is being sent. Now that we use `headlong`, we cannot send a transaction with such an invalid value, so the same error is handled on preCheck before sending the transaction. This makes the test always to FAIL. So this test case, now seems obsolete.

**Related issue(s)**: #3077 

Fixes #3077 

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilVerbs.java [hunk 1]
```diff
@@ -1310,30 +1301,28 @@ public Tuple build() {
...
-            this.tokenTransferLists = Tuple.singleton(tokenTransferLists);
+            this.tokenTransferLists = tokenTransferLists;
             return this;
         }
 
-        public Tuple build() {
-            return tokenTransferLists;
+        public Object build() {
+            return (Object) tokenTransferLists;
         }
```
Comment (georg-getz) [inline test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilVerbs.java:1313]: Why the cast here? Is there something wrong with returning `Tuple[]`?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/autorenew/GracePeriodRestrictionsSuite.java [hunk 2]
```diff
@@ -79,17 +80,17 @@ public static void main(String... args) {
...
+                    //                    payerRestrictionsEnforced(),
+                    //                    cryptoTransferRestrictionsEnforced(),
+                    //                    tokenMgmtRestrictionsEnforced(),
+                    //                    cryptoAndContractDeleteRestrictionsEnforced(),
+                    //                    treasuryOpsRestrictionEnforced(),
+                    //                    tokenAutoRenewOpsEnforced(),
+                    //                    topicAutoRenewOpsEnforced(),
+                    //                    cryptoUpdateRestrictionsEnforced(),
+                    //                    gracePeriodRestrictionsSuiteCleanup(),
                 });
```
Comment (ivanparnarev) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/autorenew/GracePeriodRestrictionsSuite.java:94]: Are these tests commented for test purposes or are affected by the change?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/SStoreSuite.java [hunk 3]
```diff
@@ -64,12 +64,12 @@ public static void main(String... args) {
...
-                    multipleSStoreOpsSucceed(),
+                    //                    setupAppProperties(),
+                    //                    multipleSStoreOpsSucceed(),
                     benchmarkSingleSetter(),
-                    childStorage(),
-                    temporarySStoreRefundTest(),
-                    cleanupAppProperties()
+                    //                    childStorage(),
+                    //                    temporarySStoreRefundTest(),
+                    //                    cleanupAppProperties()
```
Comment (ivanparnarev) [inline test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/SStoreSuite.java:72]: Are these tests commented for test purposes or are affected by the change?

File: test-clients/src/main/java/com/hedera/services/bdd/suites/autorenew/GracePeriodRestrictionsSuite.java (hunk 2 above)
Comment (IvanKavaldzhiev) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/autorenew/GracePeriodRestrictionsSuite.java]: Forgot to uncomment them. FIXED now

File: test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/SStoreSuite.java (hunk 3 above)
Comment (IvanKavaldzhiev) [reply test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/SStoreSuite.java]: DONE

File: test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilVerbs.java (hunk 1 above)
Comment (georg-getz) [reply test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/UtilVerbs.java]: Function return type could be changed to `Tuple[]` as well.

**Review by stoqnkpL — APPROVED** [review]
Very nice improvements and clean-up!

**Review by tinker-michaelj — APPROVED** [review]
Looks great, tyvm @IvanKavaldzhiev!

**Review by stoqnkpL — APPROVED** [review]

**Review by nathanklick — APPROVED** [review]

**Pull request merged**